diff --git a/src/player.cpp b/src/player.cpp
index 91f2a7a..a85fd12 100644
--- a/src/player.cpp
+++ b/src/player.cpp
@@ -94,10 +94,15 @@ int main(int argc, char **argv)
         copy(istream_iterator<string>(iss),
                 istream_iterator<string>(),
                 back_inserter(tokens));
-        if (tokens.size() != 3) {
+	for (unsigned i=3; i < tokens.size(); ++i) {
+	    tokens[2] += " " + tokens[i];
+	}
+        if (tokens.size() < 3) {
             cerr << "Bad line: " << counter << ": '" << line << "'" << endl;
             fncs::die();
-        }
+        } else {
+	    tokens.resize(3);
+	}
 
         /* convert first token into fncs::time */
         {
